packages/app-desktop/gui/NoteEditor/NoteBody/CodeMirror/utils/useCursorUtils.ts

```diff
@@ -75,6 +75,12 @@ export default function useCursorUtils(CodeMirror: any) {
 		// Batches the insert operations, if this wasn't done the inserts
 		// could potentially overwrite one another
 		this.operation(() => {
+
+			let num = 0;
+			if (string1.match(/^\d/)) {
```

Comment:
Instead of this logic here, you should use [`olLineNumber`](https://github.com/laurent22/joplin/blob/dev/packages/lib/markdownUtils.ts#L83) from the markdownUtils module.
Also if you move this logic inside the `selectedStrings` (after line 84) for loop then you'll get the correct counting behaviour if the user has multiple selections.

Reply:
Hello! I've used olLineNumber and move this logic inside of the for loop. Thank you for your feedback.

Reply:
Looks good!